packages/oauth/jwk/src/key.ts

```diff
@@ -84,6 +111,54 @@ export abstract class Key<J extends Jwk = Jwk> {
     return Object.freeze(Array.from(jwkAlgorithms(this.jwk)))
   }
 
+  matches(opts: KeyMatchOptions): boolean {
+    // Optimization: Empty string or empty array will not match any key
+    if (opts.alg?.length === 0) return false
+    if (opts.kid?.length === 0) return false
+
+    if (opts.usage) {
+      if (this.ops) {
+        if (opts.usage === 'verify' && this.ops.includes('sign')) {
+          // allowed
+        } else if (opts.usage === 'decrypt' && this.ops.includes('encrypt')) {
+          // allowed
+        } else if (!this.ops.includes(opts.usage)) {
+          return false // no match
+        }
+      }
```

Comment:
When ops is defined, only matching ops should be allowed. The provided key should contain both `verify` and `sign` if that's their intended use.

```suggestion
      if (this.ops && !this.ops.includes(opts.usage)) {
        return false // no match
      }
```